corehq/apps/events/models.py

```diff
@@ -47,17 +67,143 @@ class Meta:
             models.Index(fields=("manager_id",)),
         )
 
-    @classmethod
-    def by_domain(cls, domain, most_recent_first=False):
-        if most_recent_first:
-            return cls.objects.filter(domain=domain).order_by('start_date')
-        return cls.objects.filter(domain=domain)
-
-    def save(self):
+    def save(self, expected_attendees=None):
         if not self.event_id:
             self.event_id = uuid.uuid4().hex
-        super(Event, self).save()
+        event = super(Event, self).save()
+
+        if expected_attendees is not None:
+            self._update_attendees(expected_attendees)
+
+        return event
 
     @property
     def status(self):
         return self.attendee_list_status
+
+    @property
+    def attendees(self):
+        return Attendee.objects.get_by_event_id(self.event_id, self.domain)
+
+    def _update_attendees(self, attendees_case_ids):
+        current_attendees_ids = Attendee.objects.get_by_event_id(
+            self.event_id,
+            self.domain,
+            only_ids=True
+        )
+
+        attendees_to_assign, attendees_to_unassign = find_difference(
+            current_attendees_ids,
+            attendees_case_ids
+        )
+
+        self._assign_attendees(list(attendees_to_assign))
+        self._unassign_attendees(list(attendees_to_unassign))
+
+    def _assign_attendees(self, attendees_case_ids):
+        if not attendees_case_ids:
+            return
+
+        # How to make this idempotent?
+        domain = self.domain
+        event_id = self.event_id
+
+        for parent_case_id in attendees_case_ids:
+            case_args = {
+                'domain': domain,
+                'properties': {'event_id': event_id},
+            }
+            index_args = {
+                'parent_case_id': parent_case_id,
+                'identifier': case_index_event_identifier(event_id),
+            }
+            create_case_with_case_type(
+                case_type=Attendee.EVENT_ATTENDEE_CASE_TYPE,
+                case_args=case_args,
+                index=index_args,
+            )
+
+    def _unassign_attendees(self, attendees_case_ids):
+        """
+        This method deletes the indices and cases linking the domain attendees to the event
+        """
+        if not attendees_case_ids:
+            return
+
+        extension_cases_ids = CommCareCaseIndex.objects.get_extension_case_ids(
+            self.domain,
+            attendees_case_ids
+        )
+        if not extension_cases_ids:
```

Comment:
When this method has executed once and is called again, `extension_cases_ids` will be empty (as expected). This check simply returns since nothing is left to do.